### PR TITLE
[Carbondata-3643] Insert array('')/array() into Struct<Array> column …

### PR DESCRIPTION
**Pull request by marchpure**
…will result in array(null), which is inconsist with Parquet

Modification reason: Result is incorrect when Inserting Select From a Parquet Table with a Struct with array('')/array(, The result shouldn't be array(null), while parquet results in array('') or array().

Modification content: When the input value is Struct(""), the StructParserImpl handle the EMPTY STRING

 ### Why is this PR needed?
Insert array('')/array() into Struct<Array> column will result in array(null), which is inconsist with Parquet. The corrct result should be array('') or array().

 ### What changes were proposed in this PR?
When the input value is Struct("")/Struct(Array())/Struct(Map()), the StructParserImpl return the EMPTY STRING, insteading of return NULL.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1369/

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1379/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1389/

**Comment by jackylk** [comment]
please modify the PR description according to the template

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1371/

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1381/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1391/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1375/

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1385/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1396/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1382/

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1392/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1404/

File: processing/src/main/java/org/apache/carbondata/processing/loading/parser/impl/StructParserImpl.java [hunk 1]
```diff
@@ -59,6 +59,14 @@ public StructObject parse(Object data) {
           }
           return new StructObject(array);
         }
+      } else if (value.isEmpty()) {
```
Comment (ajantha-bhat) [inline processing/src/main/java/org/apache/carbondata/processing/loading/parser/impl/StructParserImpl.java]: I just checked the code , **Need to handle the same for `MapParserImpl` also**. where as **ArrayParserImpl** is already handled this.

**Pull request closed without merging**